src/util.rs

```diff
@@ -85,19 +121,56 @@ pub fn get_active_network_interface_gateways() -> std::io::Result<Vec<IpAddr>> {
     Ok(addrs)
 }
 
-pub(crate) fn set_interface_dns_servers(interface: GUID, dns: &[IpAddr]) -> std::io::Result<()> {
+pub(crate) fn set_interface_dns_servers(interface: GUID, dns: &[IpAddr]) -> crate::Result<()> {
     // format L"1.1.1.1,8.8.8.8", or L"1.1.1.1 8.8.8.8".
     let dns = dns.iter().map(|ip| ip.to_string()).collect::<Vec<_>>().join(",");
     let dns = dns.encode_utf16().chain(std::iter::once(0)).collect::<Vec<_>>();
 
     let settings = DNS_INTERFACE_SETTINGS {
         Version: DNS_INTERFACE_SETTINGS_VERSION1,
         Flags: DNS_SETTING_NAMESERVER as _,
-        NameServer: PWSTR(dns.as_ptr() as _),
-        ..DNS_INTERFACE_SETTINGS::default()
+        NameServer: dns.as_ptr() as _,
+        Domain: std::ptr::null_mut(),
+        SearchList: std::ptr::null_mut(),
+        RegistrationEnabled: 0,
+        RegisterAdapterName: 0,
+        EnableLLMNR: 0,
+        QueryAdapterName: 0,
+        ProfileNameServer: std::ptr::null_mut(),
     };
 
-    unsafe { SetInterfaceDnsSettings(interface, &settings as *const _)? };
+    // unsafe { SetInterfaceDnsSettings(interface, &settings as *const _) }
+
+    type TheFn = unsafe extern "system" fn(interface: GUID, settings: *const DNS_INTERFACE_SETTINGS) -> WIN32_ERROR;
```

Comment:
Anything stopping us from using [this function](https://docs.rs/windows-sys/latest/windows_sys/Win32/NetworkManagement/IpHelper/fn.SetInterfaceDnsSettings.html)? We definitely don't want to load / unload `iphlpapi.dll` every time someone changes dll settings